### PR TITLE
NIN-10: Add Dev Environment.

### PR DESCRIPTION
**Pull request by shakedlokits**
Added dev environment for all services and providers.

### CODE REVIEW

File: Caddyfile [hunk 1]
```diff
@@ -1,6 +1,8 @@
 0.0.0.0
 proxy / registration-provider:3000
-proxy /team teams-provider:3000
+proxy /teams teams-provider:3000 {
+    without /teams
```
Comment (wildeyes) [inline Caddyfile:4]: So no index? cause we need it

File: providers/registration/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,17 @@
 FROM node:10
 
 # Create app directory
-WORKDIR /usr/src/app
+WORKDIR /app
```
Comment (wildeyes) [inline providers/registration/Dockerfile:4]: 👍

File: providers/teams/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
```
Comment (wildeyes) [inline providers/teams/pages/index.tsx:1]: why is this here?

**Review by wildeyes — APPROVED** [review]

File: Caddyfile (hunk 1 above)
Comment (shakedlokits) [reply Caddyfile:4]: What do you mean by index? and why do we need it?

File: providers/teams/pages/index.tsx (hunk 3 above)
Comment (shakedlokits) [reply providers/teams/pages/index.tsx:1]: Historical reasons once upon a time react used to import some globals, not sure it's still relevant but my IDE still bitches about it. If it's not relevant you could remove it in your tickets on teams provider.

File: providers/registration/Dockerfile (hunk 2 above)
Comment (shakedlokits) [reply providers/registration/Dockerfile:4]: 🎉

File: Caddyfile (hunk 1 above)
Comment (wildeyes) [reply Caddyfile:4]: So why teams and no team again?
